src/Avalonia.Base/Data/Core/BindingExpression.cs

```diff
@@ -84,11 +85,36 @@ public BindingExpression(ExpressionObserver inner, Type targetType)
 
             _inner = inner;
             _targetType = targetType;
-            Converter = converter;
-            ConverterParameter = converterParameter;
             _fallbackValue = fallbackValue;
             _targetNullValue = targetNullValue;
             _priority = priority;
+
+            Converter = converter;
+
+            if (converterParameter is InstancedBinding binding)
+            {
+                if (binding.Mode == BindingMode.TwoWay)
+                {
+                    throw new InvalidOperationException("Two way binding to converter parameter is not supported");
+                }
+
+                _converterParameterSubscription = binding.Observable?.Subscribe(cp =>
+                {
+                    _converterParameter = new WeakReference<object?>(cp);
+
+                    if (_value != null &&
+                        _value.TryGetTarget(out var currentValue))
+                    {
+                        //Force update binding to use new converter parameter
+                        _innerListener?.OnNext(currentValue);
```

Comment:
This entire part in general feels hacky, but in particular this line: none of the binding infrastructure is built with the assumption that the `ConverterParameter` can change and so I'm worried that forcing the binding to update like this will not be reliable in all cases.

Reply:
I didn't check the code but it should be ok to call the listener observable with the same value multiple times? After all, the listener of this binding expression cannot tell if the notification came from an actual binding update or was produced manually and they need to handle it regardless.